pypeit/core/flux.py

```diff
@@ -3,209 +3,592 @@
+            logfit2_bkpt, _ = bset_log1.value(bset1.breakpoints)
+            plt.figure(1)
+            plt.plot(wave_obs, residual, drawstyle='steps-mid', color='black', label='residual')
+            plt.plot(wave_obs, resid_fit, color='cornflowerblue', label='resid_fit')
+            plt.plot(wave_obs[~masktot], residual[~masktot], '+', color='red', markersize=5.0, label='masked residual')
+            plt.plot(wave_obs[~masktot], resid_fit[~masktot], '+', color='red', markersize=5.0, label='masked resid_fit')
+            plt.plot(init_breakpoints, logfit2_bkpt, '.', color='green', markersize=4.0, label='breakpoints')
+            plt.plot(wave_obs, 1.0 / np.sqrt(residual_ivar), color='orange', label='sigma')
+            plt.legend()
+            plt.xlabel('Wavelength [ang]')
+            plt.ylim(-0.1, 0.1)
+            plt.title('2nd Bspline fit')
+            plt.show()
+    else:
+        bset_log1 = bset1.copy()
+
+    # Create sensitivity function
+    newlogfit, _ = bset_log1.value(wave_obs)
+    sensfit = np.power(10.0, 0.4 * np.maximum(np.minimum(newlogfit, MAGFUNC_MAX), MAGFUNC_MIN))
+
+    sensfit[~magfunc_mask] = 0.0
+
+    if debug:
+
+        # Check for calibration
+        plt.figure(1)
+        plt.plot(wave_obs, sensfunc, drawstyle='steps-mid', color='black', label='sensfunc')
+        plt.plot(wave_obs, sensfit, color='cornflowerblue', label='sensfunc fit')
+        plt.plot(wave_obs[~masktot], sensfunc[~masktot], '+', color='red', markersize=5.0, label='masked sensfunc')
+        plt.plot(wave_obs[~masktot], sensfit[~masktot], '+', color='red', markersize=5.0, label='masked sensfuncfit')
+        plt.legend()
+        plt.xlabel('Wavelength [ang]')
+        plt.ylim(0.0, 100.0)
+        plt.show()
+
+    # Check quality of the fit
+    absdev = np.median(np.abs(sensfit / modelfit1 - 1))
     msgs.info('Difference between fits is {:g}'.format(absdev))
 
+    # Check for residual of the fit
+    if debug:
+
+        # scale = np.power(10.0, 0.4 * sensfit)
+        flux_cal = flux_obs * sensfit
+        ivar_cal = ivar_obs / sensfit ** 2.
+
+        plt.rcdefaults()
+        plt.rcParams['font.family']= 'times new roman'
+        plt.figure(figsize=(11, 8.5))
+        plt.clf()
+        plt.plot(wave_obs,flux_cal, label='Calibrated Spectrum')
+        plt.plot(wave_obs,flux_std, label='Model')
+        plt.plot(wave_obs,np.sqrt(1/ivar_cal))
+        plt.legend()
+        plt.xlabel('Wavelength [ang]')
+        plt.ylabel('Flux [erg/s/cm2/Ang.]')
+        plt.ylim(0,np.median(flux_std)*2.5)
+        plt.show()
+        plt.close()
+
+
+
+
     # QA
     msgs.work("Add QA for sensitivity function")
+    qa_bspline_magfit(wave_obs, bset_log1, magfunc, masktot)
+    
+    
+    """
+    bspline_magfit_new_qa(wave_obs, magfunc, logfit1,
+                          newlogfit, bset1.breakpoints,
+                          outfile=None, title=None)
+    """
 
-    return tck_log1
+    return bset_log1
```

Comment:
Can you include a PNG of this new QA in the pull request notes?
thanks.

Reply:
We don't understand how to write out QA in PypeIt. Maybe you can help us with this.